File: sample/src/main/java/com/nathaniel/sample/SampleAdapter.java
package com.nathaniel.sample;

import com.nathaniel.adapter.adapter.BaseRecyclerAdapter;
import com.nathaniel.adapter.adapter.BaseViewHolder;

import java.util.List;

/**
 * @author Nathaniel
 */
public class SampleAdapter extends BaseRecyclerAdapter<String> {
    public SampleAdapter(int layoutId, List<String> dataList) {
        super(layoutId, dataList);
    }

    @Override
    public void bindDataToView(BaseViewHolder holder, String data) {
        holder.setText(com.nathaniel.sample.R.id.item_text_tv, data);

        holder.addClickListener(R.id.item_button_tv);
    }
}

File: sample/src/main/java/com/nathaniel/sample/SampleAdapter.kt
package com.nathaniel.sample

import com.nathaniel.adapter.adapter.BaseRecyclerAdapter
import com.nathaniel.adapter.adapter.BaseViewHolder

/**
 * @author Nathaniel
 */
class SampleAdapter(layoutId: Int, dataList: List<String?>?) : BaseRecyclerAdapter<String?>(layoutId, dataList) {
    override fun bindDataToView(holder: BaseViewHolder, data: String) {
        holder.setText(R.id.item_text_tv, data)
        holder.addClickListener(R.id.item_button_tv)
    }
}